### ISSUE TITLE
p::s::Tria -- consistency of cells marked for refinement

### ISSUE THREAD

**Issue by davydden** [dealii/dealii#4176]
we need to either add asserts to check that same cells are marked for refinement on each MPI core, or support the case
```
if (cell->is_locally_owned())
   cell->set_refine_flag ();
```
which is how `p::d::Tria` is used. I am in favour of the latter so that one can easily switch between the two. Failing test can be found in https://github.com/dealii/dealii/pull/4167

**Comment by bangerth** [comment]
Yes, this seems like the right direction. Every processor *may* mark all cells, but ultimately what matters is only how those cells are marked that this processor *owns*.

How do you intend to communicate between processors which cells to refine? Create a bit mask of length `n_active_cells()` on each processor, do an `MPI_Allreduce` with bit-or?

**Comment by davydden** [comment]
> How do you intend to communicate between processors which cells to refine? Create a bit mask of length n_active_cells() on each processor, do an MPI_Allreduce with bit-or?

i think that's ultimately the only way to do this as we have a shared Triangulation underneath so we need to synchronise all marked cells on different MPI cores before we go on with the refinement.

**Comment by drwells** [comment]
This check is in #20099.
